src/engine/pieces/piece.ts

```diff
@@ -16,5 +29,65 @@ export default class Piece {
     public moveTo(board: Board, newSquare: Square) {
         const currentSquare = board.findPiece(this);
         board.movePiece(currentSquare, newSquare);
+        this.active = true;
+    }
+
+    public isValidMove(board: Board, nextSquare: Square) {
+
+        let pieceOnBoard = board.getPiece(nextSquare)
+        if (pieceOnBoard === undefined)
+            return SquareState.EMPTY_SQUARE;
+
+        if ((this.player != pieceOnBoard.player) && (pieceOnBoard.pieceType != PieceType.KING))
+            return SquareState.CAPTURABLE_PIECE_PRESENT;
+
+        return SquareState.UNCAPTURABLE_PIECE_PRESENT;
+    }
+
+    private getDirectionalMoves(board: Board, directionChanges: Array<PositionChange>)
+    {
+        let validMoves = new Array(0);
```

Comment:
```suggestion
        const validMoves = [];
```

it's more common practice to define arrays like this, also would be good to add a type to `validMoves`

Arrays are objects which are always mutable in Typescript so we would always define them as a const